server/models/user.js

```diff
@@ -48,9 +61,9 @@ userSchema.methods.isMemberOfAny = function isMemberOfAny(groupIds) {
   return groupIds.some(this.isMemberOf, this);
 };
 
-userSchema.pre('save', function (next) {
+userSchema.pre('save', function preSave(next) {
   // Add user's hard-coded groups
-  if (this.groups.length === 0) {
+  if (this.groups && this.groups.length === 0) {
```

Comment:
Was causing problems when groups weren't included in the projection I was saving if I didn't do this check